File: fortran to python conversion/dependencies/subroutine RMScalc.py
import math

def RMScalc(co, nats, dir, ip):
	"""

	Modified from the original to the calculate the sd of the atom to axis distance

	***
	Parameters
	co: the input coordinates
	nats: the number of atoms
	ip: the coordinates of the initial point
	dir: the axis vector

	Returns:
	the root mean squared of the set of point
	fp: the coordinates of the final point

	"""
	tmp = []
	# calculate rms deviations
	sum = 0
	sum2 = 0
	for i in range(nats):
		tmp = co[0][i] - ip
		tmp = dvproj(dir, tmp)
		fp = ip + tmp
		dd= ddistsq(co[0][i], fp)
		sum+= math.sqrt(dd)
		sum2+=dd
	return sqrt(sum2/nats - (sum/nats)**2),fp

File: subroutines/selected/dependencies/subroutine parlsq dependencies/subroutine RMScalc.f
      subroutine RMScalc(co,nats,dir,ip,fp,rms)
      implicit real*8(a-h,o-z)
      real*8 co(3,nats) !the input coordinates
c     Modified from the original to calculate the sd of the atom to axis dist
      integer nats              !number of atoms
      real*8 ip(3)              !the coordinates of the initial point
      real*8 fp(3)              !the coordinates of the final point
      real*8 dir(3)             !the axis vector

      real*8 tmp(3)

c     calculate rms deviations
      rms=0.0
      sum=0.0
      sum2=0.0
      do i=1,nats
         call dvdif (co(1,i),ip,tmp)
         call dvproj(dir,tmp,tmp)
         call dvsum(ip,tmp,fp)
         dd=ddistsq(co(1,i),fp)
         sum=sum+sqrt(dd)
         sum2=sum2+dd
      end do
      rms=sqrt(sum2/float(nats)-(sum/float(nats))**2)
      end
